src/vcpkg/registries.cpp

```diff
@@ -1208,6 +1308,25 @@ namespace vcpkg
             });
     }
 
+    ExpectedL<std::map<std::string, std::string>> get_builtin_port_git_objects(const VcpkgPaths& paths)
+    {
+        const auto& git = paths.get_git_impl(stdout_sink);
+        GitLsTreeOptions opts = {};
+        opts.dirs_only = true;
+        auto maybe_files = git.ls_tree(paths.git_builtin_config(), "HEAD:ports/", opts);
```

Comment:
End-to-end tests are failing due to this hardcoded `"HEAD:ports/"`. We should make this use the value of `--x-builtin-ports-root`.